mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -141,22 +143,63 @@ public void serviceShouldSaveAttachmentInAFolderWhenPatternMatch() throws Messag
+                .setProperty("mimeType", "text/calendar")
+                .setProperty("remove", "matched")
+                .build();
+        Mailet mailet = new StripAttachment();
+        mailet.init(mci);
+
+        MimeMultipart multiPart = new MimeMultipart();
+        MimeBodyPart textPart = new MimeBodyPart();
+        textPart.setText("simple text");
+        multiPart.addBodyPart(textPart);
+        String expectedFileName = "10.ical";
+        multiPart.addBodyPart(createAttachmentBodyPart("content", expectedFileName, Optional.of("text/calendar")));
+        multiPart.addBodyPart(createAttachmentBodyPart("other content", "11.ical", ABSENT_MIME_TYPE));
+        multiPart.addBodyPart(createAttachmentBodyPart("<p>html</p>", "index.html", Optional.of("text/html")));
+
+        MimeMessage message = mimeMessage();
+        message.setSubject("test");
+        message.setContent(multiPart);
+        message.saveChanges();
+
+        Mail mail = FakeMail.builder()
+                .mimeMessage(message)
+                .build();
+
+        mailet.service(mail);
+
+        @SuppressWarnings("unchecked")
+        List<String> removedAttachments = (List<String>) mail.getAttribute(StripAttachment.REMOVED_ATTACHMENTS_ATTRIBUTE_KEY);
+        assertThat(removedAttachments).containsOnly(expectedFileName);
+    }
+
     private MimeMessage mimeMessage() {
         return new MimeMessage(Session
                 .getDefaultInstance(new Properties()));
     }
 
-    private MimeBodyPart createAttachmentBodyPart(String body, String fileName) throws MessagingException, UnsupportedEncodingException {
-        MimeBodyPart part = createBodyPart(body);
+    private MimeBodyPart createAttachmentBodyPart(String body, String fileName, Optional<String> mimeType) throws MessagingException, UnsupportedEncodingException {
+        MimeBodyPart part = createBodyPart(body, mimeType);
         part.setDisposition("attachment");
         part.setFileName(fileName);
         return part;
     }
 
-    private MimeBodyPart createBodyPart(String body) throws MessagingException, UnsupportedEncodingException {
-        return new MimeBodyPart(new ByteArrayInputStream(
-                ("Content-Transfer-Encoding: 8bit\r\nContent-Type: application/octet-stream; charset=utf-8\r\n\r\n"
-                        + body).getBytes("UTF-8")));
+    private MimeBodyPart createBodyPart(String body, Optional<String> mimeType) throws MessagingException, UnsupportedEncodingException {
+        byte[] content = (mimeHeaders(mimeType) + body).getBytes("UTF-8");
+        return new MimeBodyPart(new ByteArrayInputStream(content));
+    }
+
+    private String mimeHeaders(Optional<String> mimeType) {
+        if (mimeType.isPresent()) {
+            return "Content-Transfer-Encoding: 8bit\r\nContent-Type: " + mimeType.get() + "; charset=utf-8\r\n\r\n";
+        }
+        return "Content-Transfer-Encoding: 8bit\r\nContent-Type: application/octet-stream; charset=utf-8\r\n\r\n";
```

Comment:
I don't like very much conditionals into tests. Cannot you simply create constants?